From 011a6d30f33b07178f9b78431764027e7c3febc7 Mon Sep 17 00:00:00 2001
From: vfxpro99 <vfxpro99@gmail.com>
Date: Fri, 18 Nov 2016 17:29:45 -0800
Subject: [PATCH] Patch for Maya detection

---
 cmake/modules/FindMaya.cmake | 17 +++++++----------
 1 file changed, 7 insertions(+), 10 deletions(-)

diff --git a/cmake/modules/FindMaya.cmake b/cmake/modules/FindMaya.cmake
index 0a742be..e07d6ca 100644
--- a/cmake/modules/FindMaya.cmake
+++ b/cmake/modules/FindMaya.cmake
@@ -29,18 +29,11 @@
 if(APPLE)
     find_path(MAYA_BASE_DIR
             ../../devkit/include/maya/MFn.h
+            include/maya/MFn.h
         HINTS
             "${MAYA_LOCATION}"
             "$ENV{MAYA_LOCATION}"
             "/Applications/Autodesk/maya2017/Maya.app/Contents"
-            "/Applications/Autodesk/maya2015/Maya.app/Contents"
-            "/Applications/Autodesk/maya2014/Maya.app/Contents"
-            "/Applications/Autodesk/maya2013.5/Maya.app/Contents"
-            "/Applications/Autodesk/maya2013/Maya.app/Contents"
-            "/Applications/Autodesk/maya2012.17/Maya.app/Contents"
-            "/Applications/Autodesk/maya2012/Maya.app/Contents"
-            "/Applications/Autodesk/maya2011/Maya.app/Contents"
-            "/Applications/Autodesk/maya2010/Maya.app/Contents"
     )
     find_path(MAYA_LIBRARY_DIR libOpenMaya.dylib
         HINTS
@@ -52,6 +45,8 @@ if(APPLE)
         DOC
             "Maya's libraries path"
     )
+    set(_PXR_CXX_DEFINITIONS "${_PXR_CXX_DEFINITIONS} -DOSMac_MachO_ -DOSMac_")
+
 elseif(UNIX)
     find_path(MAYA_BASE_DIR
             include/maya/MFn.h
@@ -150,12 +145,14 @@ find_path(MAYA_LIBRARY_DIR
 list(APPEND MAYA_INCLUDE_DIRS ${MAYA_INCLUDE_DIR})

 find_path(MAYA_DEVKIT_INC_DIR
-       GL/glext.h
+       tbb/tbb.h
+
     HINTS
         "${MAYA_LOCATION}"
         "$ENV{MAYA_LOCATION}"
         "${MAYA_BASE_DIR}"
     PATH_SUFFIXES
+        include
         ../../devkit/plug-ins/
     DOC
         "Maya's devkit headers path"
@@ -184,6 +181,7 @@ foreach(MAYA_LIB
             "${MAYA_BASE_DIR}"
         PATH_SUFFIXES
             MacOS/
+            Maya.app/Contents/MacOS/
             lib/
         DOC
             "Maya's ${MAYA_LIB} library path"
@@ -192,7 +190,6 @@ foreach(MAYA_LIB
         NO_CMAKE_SYSTEM_PATH
     )

-
     if (MAYA_${MAYA_LIB}_LIBRARY)
         list(APPEND MAYA_LIBRARIES ${MAYA_${MAYA_LIB}_LIBRARY})
     endif()
--
2.8.4 (Apple Git-73)
